File: Исходники/Глава 3. Часть 1/5. Сортировка пузырьком/Python/bubble sort.py
def bubble_sort(A: list):
    for i in range(len(A) - 1):
        for j in range(len(A) - 1):
	        if A[j] > A[j + 1]:
		        A[j], A[j + 1] = A[j + 1], A[j]
    return A

File: Исходники/Глава 3. Часть 1/5. Сортировка пузырьком/C++/Продуктовый/bubble sort.cpp
template<typename T>
bool comp(T firstData, T secondData)
{
	return firstData > secondData;
}

template <typename T arr>
T* bubble_sort(T *arr, int size, bool comp)
{
    int i = 0;
	bool repeat = true;
    while (repeat)
    {
	    repeat = false;
        for (j = 0; j < n - i - 2; j++)
        {
	        if (comp(a[j], a[j + 1]))
	        {
		        std::swap(a[j], a[j + 1]);
		        repeat = true;
	        }
        }
	    i++; 
    }
}